File: simplerpc/src/main/java/com/springmarker/simplerpc/protocol/net/http/pojo/HttpExchangeRequest.java
package com.springmarker.simplerpc.protocol.net.http.pojo;

import com.springmarker.simplerpc.pojo.ExchangeRequest;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

/**
 * @author Springmarker
 * @date 2018/10/24 23:46
 */
@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
public class HttpExchangeRequest extends ExchangeRequest {
    private String type;
    private int methodHashCode;
    private List<Object> paramList;
    private int needReturn;

}

File: simplerpc/src/main/kotlin/com/springmarker/simplerpc/protocol/http/pojo/HttpExchangeRequest.kt
package com.springmarker.simplerpc.protocol.http.pojo

import com.springmarker.simplerpc.pojo.ExchangeRequest

/**
 * @author Frank
 * @date 2018/10/24 23:46
 */
data class HttpExchangeRequest(
        val type: String, 
        override val clientId: Int, 
        override val id: Int, 
        override val methodHashCode: Int, 
        override val paramList: List<Any>, 
        override val needReturn: Int
) 